.tekton/stonesoup-e2e-test.yaml

```diff
@@ -0,0 +1,133 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: stonesoup-e2e-test
+  annotations:
+    pipelinesascode.tekton.dev/on-event: "[push]"
+    pipelinesascode.tekton.dev/on-target-branch: "main"
+    pipelinesascode.tekton.dev/task-1: "[.tekton/tasks/deploy-cluster.yaml]"
+    pipelinesascode.tekton.dev/task-2: "[.tekton/tasks/destroy-cluster.yaml]"
+    pipelinesascode.tekton.dev/task-3: "[.tekton/tasks/generate-cluster-name.yaml]"
+    pipelinesascode.tekton.dev/task-4: "[.tekton/tasks/stonesoup-tests.yaml]"
+    pipelinesascode.tekton.dev/task-5: "[git-clone]"
+    pipelinesascode.tekton.dev/max-keep-runs: "5"
+spec:
+  params:
+    - name: repo_url
+      value: "{{ repo_url }}"
+    - name: revision
+      value: "{{ revision }}"
+    - name: repo_owner
+      value: "{{ repo_owner }}"
+    - name: target_branch
+      value: "{{ target_branch }}"
+  timeouts:
+    pipeline: "1h30m0s"
+    finally: "0h20m0s"
+  pipelineSpec:
```

Comment:
@xinredhat Can we create separate pipeline for this and add pipelineRef here?

Reply:
@prietyc123 because the separate pipeline won't be reused in other pipelineruns, do we need to create a separate pipeline for this?

Reply:
I think maintaining the similar structure all over makes sense. WDYT? Also in near future we don't know how much stonesoup role is going to be played from our end.

Reply:
Done, I have verified the CI and it works